lms/views/api/blackboard/sync.py

```diff
@@ -11,68 +12,68 @@ def __init__(self, request):
         self.grouping_service = self.request.find_service(name="grouping")
         self.blackboard_api = self.request.find_service(name="blackboard_api_client")
 
-        self.tool_consumer_instance_guid = self.request.json["lms"][
-            "tool_consumer_instance_guid"
-        ]
-
     @view_config(
         route_name="blackboard_api.sync",
         request_method="POST",
         renderer="json",
         permission=Permissions.API,
+        schema=APISyncBlackboardSchema,
     )
     def sync(self):
-        groups = self.get_blackboard_groups()
+        params = self.request.parsed_params
+        tool_consumer_instance_guid = params["lms"]["tool_consumer_instance_guid"]
+        resource_link_id = params["assignment"]["resource_link_id"]
+        group_info = params["group_info"]
+
+        group_set_id = self.group_set(tool_consumer_instance_guid, resource_link_id)
+        course = self.get_course(
+            tool_consumer_instance_guid, params["course"]["context_id"]
+        )
 
-        self.sync_to_h(groups)
+        groups = self.get_blackboard_groups(course, group_set_id)
 
+        self.request.find_service(name="lti_h").sync(groups, group_info)
         authority = self.request.registry.settings["h_authority"]
         return [group.groupid(authority) for group in groups]
 
-    def get_blackboard_groups(self):
+    def get_blackboard_groups(self, course, group_set_id):
         lti_user = self.request.lti_user
-        group_set_id = self.group_set()
-        course_id = self.request.json["course"]["context_id"]
-        course = self.get_course(course_id)
 
         if lti_user.is_learner:
             user = self.request.find_service(UserService).get(
-                self.request.find_service(name="application_instance").get_current(),
+                course.application_instance,
```

Comment:
Using the relationship here instead of another separate call to `ai_service.get_current()`